clients/src/main/java/org/apache/kafka/common/internals/KafkaFutureImpl.java

```diff
@@ -104,7 +104,7 @@ R await(long timeout, TimeUnit unit)
                         wrapAndThrow(exception);
                     if (done)
                         return value;
-                    if (delta > waitTimeMs) {
+                    if (delta >= waitTimeMs) {
```

Comment:
Since we are now handling equality, I think we can change the line above to initialize `waitTimeMs` to this:
```java
long waitTimeMs = unit.toMillis(timeout);
```
Then to test the boundary condition, we just need to verify the behavior of `await(0)`.

Reply:
Since we're pressed for time a bit to get the release out, I will go ahead and make this change when merging the PR. Hopefully it's ok.